cdap-app-templates/cdap-etl/cdap-data-streams/src/main/java/io/cdap/cdap/datastreams/DataStreamsSparkLauncher.java

```diff
@@ -118,12 +115,33 @@ public void initialize() throws Exception {
     // without this, stopping will hang on machines with few cores.
     sparkConf.set("spark.rpc.netty.dispatcher.numThreads", String.valueOf(numSources + 2));
 
-    sparkConf.set("spark.executor.instances", String.valueOf(numSources + 2));
     sparkConf.setMaster(String.format("local[%d]", numSources + 2));
 
     if (spec.isUnitTest()) {
       sparkConf.setMaster(String.format("local[%d]", numSources + 1));
     }
+
+    // override defaults with any user provided engine configs
+    int minExecutors = numSources + 1;
```

Comment:
should this be `numSources + 2`?

Reply:
it was +2 before as a default to allow 2 executors to process the data, but the actual minimum is +1.

I think I should change the logic a little so that the default stays the same unless it's in these properties. Will make the change.